### PR TITLE
Force push handle folders

### PR DESCRIPTION
**Pull request by yurybond**
I have added logic to handle folders with force push. It can be helpful if you want to push full aura component. For instance

`force push -f src/aura/MyComponent/`

### CODE REVIEW

**Comment by cwarden** [comment]
Thanks, @yurybond!  I'll check this out soon.

**Comment by yurybond** [comment]
Did you have a chance to review this pull request?

File: lib/deploy.go [hunk 1]
```diff
@@ -15,17 +16,49 @@ import (
 // and then deploys the package to salesforce
 func PushByPaths(fpaths []string, byName bool, namePaths map[string]string, opts *ForceDeployOptions) {
 	pb := NewPushBuilder()
-
+	fmt.Println("directory")
```
Comment (dcarroll) [inline lib/deploy.go]: Is this just for debugging?

File: lib/deploy.go [hunk 2]
```diff
@@ -15,17 +16,49 @@ import (
...
-		} else {
-			// Store paths by name for error messages
-			namePaths[name] = fpath
+			continue
+		}
+
+		mode := fi.Mode()
+
+		if mode.IsDir() {
+			fmt.Println("directory")
```
Comment (dcarroll) [inline lib/deploy.go]: Same question here, is this just for debugging?

**Comment by cwarden** [comment]
There's a bunch of code to handle pushing aura components in `command/pushAura.go`.  Is there more than just walking the directory that we need to be doing when pushing aura components by path?

**Comment by yurybond** [comment]
Hi @cwarden, thank you for getting back. There is a lot more than just for aura, aura component deployment is just an example.

1. There was a comment `// TODO: check for folder, if a folder, add all files in it` so it supposed to be added somewhere in future 
2. Sometimes we need to push all folder content  - for instance all data model from `objects/` 
3. Sometimes we need to deploy AuraCompoment along with other metadata in a single shot - for instance new custom field + apex-controller + aura component.

P.S. Frankly, we use Force CLI for partial deployments in our CI, so having this functionality will make our life much easier.

File: lib/deploy.go (hunk 1 above)
Comment (yurybond) [reply lib/deploy.go]: Yea, will be removed.

Comment (yurybond) [reply lib/deploy.go]: @dcarroll  these debugging stuff have been removed by next commit.

File: lib/deploy.go [hunk 3]
```diff
@@ -15,17 +16,47 @@ import (
...
+		if mode.IsDir() {
+			files, err := ioutil.ReadDir(fpath)
+			if err != nil {
+				fmt.Println(err)
+				badPaths = append(badPaths, fpath)
+			}
+
+			for _, f := range files {
+				if f.IsDir() {
+					continue
```
Comment (cwarden) [inline lib/deploy.go]: I think pushing a directory should always be recursive.  For example, `force push src/` and `force push src/reports/` should work.

File: lib/deploy.go [hunk 4]
```diff
@@ -15,17 +16,47 @@ import (
...
 			badPaths = append(badPaths, fpath)
-		} else {
-			// Store paths by name for error messages
-			namePaths[name] = fpath
+			continue
+		}
+
+		mode := fi.Mode()
+		//If path provided is dir we are adding all containing files to deployment
+		if mode.IsDir() {
```
Comment (cwarden) [inline lib/deploy.go]: How about refactoring this branch to make use of a new function for adding a directory to the `PackageBuilder`? Maybe:
```
func (pb *PackageBuilder) AddDirectory(fpath string) (namePaths map[string]string, badPaths []string)
````

Comment (yurybond) [reply lib/deploy.go]: Reasonable, I will update code soon

**Comment by yurybond** [comment]
@cwarden  I have done the changes you proposed. Could you please review again?

**Comment by cwarden** [comment]
This is a great addition.  Thanks, @yurybond.  I've squashed and merged your changes.

**Pull request closed without merging**